modules/table/src/integrationTest/java/org/apache/ignite/internal/table/ItEstimatedSizeTest.java

```diff
@@ -0,0 +1,182 @@
+ */
+public class ItEstimatedSizeTest extends ClusterPerTestIntegrationTest {
+    private static final String TEST_ZONE_NAME = "TestZone";
+
+    private static final String TEST_TABLE_NAME_PREFIX = "Test_";
+
+    private static final long NUM_ROWS = 100;
+
+    private static final String[] ALL_STORAGE_PROFILES = {
+            DEFAULT_TEST_PROFILE_NAME,
+            // TODO: https://issues.apache.org/jira/browse/IGNITE-22616
+            // DEFAULT_AIPERSIST_PROFILE_NAME,
+            // TODO: https://issues.apache.org/jira/browse/IGNITE-22616
+            // DEFAULT_AIMEM_PROFILE_NAME,
+            // TODO: https://issues.apache.org/jira/browse/IGNITE-22617
+            // DEFAULT_ROCKSDB_PROFILE_NAME
+    };
+
+    @BeforeEach
+    void setUp() {
+        executeSql(String.format(
+                "CREATE ZONE %s WITH "
+                        + "REPLICAS=%d, "
+                        + "DATA_NODES_AUTO_ADJUST_SCALE_UP=%d, "
+                        + "DATA_NODES_AUTO_ADJUST_SCALE_DOWN=%d, "
+                        + "STORAGE_PROFILES='%s'",
+                TEST_ZONE_NAME,
+                initialNodes(),
+                IMMEDIATE_TIMER_VALUE,
+                IMMEDIATE_TIMER_VALUE,
+                String.join(", ", ALL_STORAGE_PROFILES)
+        ));
+    }
+
+    @Test
+    void testEstimatedSize() {
+        for (String profile : ALL_STORAGE_PROFILES) {
+            String tableName = createTableWithData(profile);
+
+            assertThat(tableSize(tableName), willBe(NUM_ROWS));
+
+            for (int i = 0; i < NUM_ROWS / 2; i++) {
+                executeSql(String.format("DELETE FROM %s WHERE key = %d", tableName, i));
+            }
+
+            assertThat(tableSize(tableName), willBe(NUM_ROWS / 2));
+
+            executeSql(String.format("DELETE FROM %s", tableName));
+
+            assertThat(tableSize(tableName), willBe(0L));
+        }
+    }
+
+    @Test
+    void testEstimatedSizeAfterScaleUp() throws InterruptedException {
+        for (String profile : ALL_STORAGE_PROFILES) {
+            String tableName = createTableWithData(profile);
+
+            assertThat(tableSize(tableName), willBe(NUM_ROWS));
+        }
+
+        cluster.startNode(initialNodes());
+
+        for (String profile : ALL_STORAGE_PROFILES) {
+            waitForRebalance(initialNodes() + 1);
+
+            assertThat(tableSize(tableName(profile)), willBe(NUM_ROWS));
+        }
+    }
+
+    @Test
+    void testEstimatedAfterScaleDown() throws InterruptedException {
+        for (String profile : ALL_STORAGE_PROFILES) {
+            String tableName = createTableWithData(profile);
+
+            assertThat(tableSize(tableName), willBe(NUM_ROWS));
+        }
+
+        cluster.stopNode(initialNodes() - 1);
+
+        for (String profile : ALL_STORAGE_PROFILES) {
+            waitForRebalance(initialNodes() - 1);
+
+            assertThat(tableSize(tableName(profile)), willBe(NUM_ROWS));
+        }
+    }
+
```

Comment:
- I'd rather add test that will check whether pending transactions (that not yet committed) effect the size.

Reply:
I already have such tests on the Partition Storage level, I think writing full integration tests for that is not needed

Reply:
Yep, makes sense.